linetools/spectra/utils.py

```diff
@@ -292,11 +304,11 @@ def rebin(spec, new_wv, do_sig=False, all=False, **kwargs):
         new_sig = None
 
     # update continuum
-    if spec.co_is_set:
-        x, y = spec._get_contpoints()
-        new_co = spec._interp_continuum(x, y, new_wv)
-    else:
-        new_co = None
+    new_co = None
+    if do_co:
+        if spec.co_is_set:
+            x, y = spec._get_contpoints()
+            new_co = spec._interp_continuum(x, y, new_wv)
```

Comment:
Could we just rebin the continuum using the continuum (`spec.co`) itself instead of the knots, which we will no longer be storing in the header?